src/util/XDRCereal.h

```diff
@@ -34,21 +34,28 @@ cereal_override(cereal::JSONOutputArchive& ar, const xdr::opaque_array<N>& s,
                  field);
 }
 
-// We still need one explicit composite-container override because cereal
-// appears to process arrays-of-arrays internally, without calling back through
-// an NVP adaptor.
-template <uint32_t N>
-void
-cereal_override(cereal::JSONOutputArchive& ar,
-                const xdr::xarray<stellar::Hash, N>& s, const char* field)
+template <typename T>
+std::enable_if_t<xdr::xdr_traits<T>::is_container>
+cereal_override(cereal::JSONOutputArchive& ar, T const& t, const char* field)
 {
-    std::vector<std::string> tmp;
-    for (auto const& h : s)
+    // setNextName, startNode, and finishNode create a new sub-object
```

Comment:
You might add some details about the fact that `setNextName, startNode, ..., finishNode` is basically what `ar(cereal::make_nvp(....))` does.

Reply:
Updated the comment. As I was updating it, I started to think that explicitly calling prologue and epilogue might be easier for future developers to understand since it'll be more similar to `ar()`, so I went ahead and replaced them. Let me know what you think!